test/distributed/test_distributed.py

```diff
@@ -2132,6 +2132,63 @@ def test_DistributedDataParallel_SyncBatchNorm_Diff_Input_Sizes_Running_Value(se
         torch.testing.assert_allclose(running_mean, all_input_var.mean(1))
         torch.testing.assert_allclose(running_var, all_input_var.var(1))
 
+    @unittest.skipIf(BACKEND != 'nccl' and BACKEND != 'gloo',
+                     "Only Nccl & Gloo backend support DistributedDataParallel")
+    @skip_if_no_cuda_distributed
+    @skip_if_no_gpu
+    def test_DistributedDataParallel_SyncBatchNorm_Diff_Input_Sizes_gradient(self):
```

Comment:
it looks like with minor modifications to _prepare_dummy_data to handle variable batch size you should be able to reuse _test_DistributedDataParallel_SyncBatchNorm?

Reply:
I can do so. I was worried about being too intrusive, as I would need to plumb the extra scaling factor all the way through `_test_DDP_5iter` as well (scale from this line https://github.com/pytorch/pytorch/pull/36382/files/e65860367bcb89ddde44c10922b1f9bd82417e67#diff-3469d9b41904e6ed24347065234034e2R2177)

But I changed `_test_DDP_HELP` already so I'll might do it as well to keep code simpler :man_shrugging: